R/chrome.R

```diff
@@ -261,44 +261,57 @@ print_page = function(
     if (!is.null(id)) switch(
       id,
       # Command #1 received -> callback: command #2 Page.enable
-      ws$send('{"id":2,"method":"Page.enable"}'),
+      ws$send(to_json(list(id = 2, method = "Page.enable"))),
       # Command #2 received -> callback: command #3 Runtime.addBinding
-      ws$send('{"id":3,"method":"Runtime.addBinding","params":{"name":"pagedownListener"}}'),
+      ws$send(to_json(list(
+        id = 3, method = "Runtime.addBinding",
+        params = list(name = "pagedownListener")
+      ))),
       # Command #3 received -> callback: command #4 Network.Enable
-      ws$send('{"id":4,"method":"Network.enable"}'),
+      ws$send(to_json(list(id = 4, method  = "Network.enable"))),
       # Command #4 received -> callback: command #5 Page.addScriptToEvaluateOnNewDocument
-      ws$send(
-        sprintf('{"id":5,"method":"Page.addScriptToEvaluateOnNewDocument","params":{"source":"%s"}}',
-                paste0(readLines(pkg_resource('js', 'chrome_print.js')), collapse = ""))
-      ),
+      ws$send(to_json(list(
+          id = 5, method = "Page.addScriptToEvaluateOnNewDocument",
+          params = list(source = paste0(readLines(pkg_resource('js', 'chrome_print.js')), collapse = ""))
+      ))),
       # Command #5 received -> callback: command #6 Page.Navigate
-      ws$send(sprintf('{"id":6,"method":"Page.navigate","params":{"url":"%s"}}', url)),
+      ws$send(to_json(list(
+          id = 6, method= "Page.navigate", params = list(url = url)
+      ))),
       # Command #6 received - check if there is an error when navigating to url
       token$error <- msg$result$errorText,
       {
       # Command #7 received - Test if the html document uses the paged.js polyfill
       # if not, call the binding when fonts are ready
-        if (!isTRUE(msg$result$result$value))
-          ws$send('{"id":8,"method":"Runtime.evaluate","params":{"expression":"pagedownReady.then(() => {pagedownListener(\'\');})"}}')
+        if (!isTRUE(msg$result$result$value)) {
+          ws$send(to_json(list(
+            id = 8, method = "Runtime.evaluate",
+            params = list(expression = "pagedownReady.then(() => {pagedownListener(\'\');})")
```

Comment:
The backslashes before the single quotes are not really necessary since they are inside a pair of double quotes.

```suggestion
            params = list(expression = "pagedownReady.then(() => {pagedownListener('');})")
```